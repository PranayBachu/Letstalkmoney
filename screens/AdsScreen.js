import React, { useEffect } from "react";
import {
  AdMobBanner,
  AdMobInterstitial,
  PublisherBanner,
  AdMobRewarded,
  setTestDeviceIDAsync,
} from "expo-ads-admob";
import { StyleSheet, View, KeyboardAvoidingView } from "react-native";
import { Button, Input, Text, Image } from "react-native-elements";

const AdsScreen = ({ navigation }) => {
  const setTestDeviceId = async () => {
    try {
      await setTestDeviceIDAsync("EMULATOR");
    } catch (error) {
      console.log(error);
    }
  };

  const interstatial = async () => {
    try {
      await AdMobInterstitial.setAdUnitID(
        "ca-app-pub-3940256099942544/1033173712"
      ); // Test ID, Replace with your-admob-unit-id
      await AdMobInterstitial.requestAdAsync({ servePersonalizedAds: true });
      await AdMobInterstitial.showAdAsync();
    } catch (error) {
      console.log(error);
    }
  };
  const reward = async () => {
    try {
      await AdMobRewarded.setAdUnitID("ca-app-pub-3940256099942544/5224354917"); // Test ID, Replace with your-admob-unit-id
      await AdMobRewarded.requestAdAsync();
      await AdMobRewarded.showAdAsync();
    } catch (error) {
      console.log(error);
    }
  };

  useEffect(() => {
    setTestDeviceId();
  }, []);

  return (
    <View style={styles.container}>
      <View
        style={{
          alignItems: "center",
          position: "absolute",
          left: 0,
          right: 0,
          bottom: 0,
        }}
      >
        <AdMobBanner
          bannerSize="banner"
          adUnitID="ca-app-pub-3940256099942544/6300978111" // Test ID, Replace with your-admob-unit-id
          servePersonalizedAds // true or false
          //onDidFailToReceiveAdWithError={this.bannerError}
        />
      </View>

      <Button
        buttonStyle={styles.button}
        onPress={interstatial}
        title="Interstatial"
      />
      <Button buttonStyle={styles.button} onPress={reward} title="Reward" />
    </View>
  );
};

export default AdsScreen;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
    // padding: 10,
    backgroundColor: "white",
    // width: 400,
  },
  button: {
    width: 200,
    marginTop: 10,
  },
  inputContainer: {
    width: 300,
  },
});
