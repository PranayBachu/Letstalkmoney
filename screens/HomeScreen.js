import React, { useEffect, useLayoutEffect, useState } from "react";
import { StatusBar } from "expo-status-bar";
import {
	StyleSheet,
	TouchableOpacity,
	Text,
	View,
	SafeAreaView,
	ScrollView,
	Modal,
} from "react-native";
import Icon from "react-native-vector-icons/FontAwesome";

import CustomListItem from "../components/CustomListItem";
import { AntDesign, SimpleLineIcons } from "@expo/vector-icons";
import { Avatar, Button, Input } from "react-native-elements";
import { firestore, auth, firebase } from "../firebase-services";

const HomeScreen = ({ navigation }) => {
	const [chats, setChats] = useState([]);
	const [chatRoomId, setChatRoomId] = useState("");
	const [isJoining, setIsJoining] = useState(false);
	const [isModalVisible, setIsModalVisible] = useState(false);

	const handleSignOutUser = async () => {
		try {
			await auth.signOut();
		} catch (error) {
			console.log(error);
		}
	};

	const handleJoinRoom = async () => {
		try {
			setIsJoining(true);
			await firestore
				.collection("chats")
				.doc(chatRoomId)
				.update({
					participants: firebase.firestore.FieldValue.arrayUnion(
						auth.currentUser.uid
					),
				});
			setIsModalVisible(false);
		} catch (error) {
			alert(error);
		} finally {
			setIsJoining(false);
		}
	};

	useEffect(() => {
		const unsubscribe = firestore
			.collection("chats")
			.where("participants", "array-contains", auth.currentUser.uid)
			.onSnapshot((snapshot) => {
				setChats(
					snapshot.docs.map((doc) => ({
						id: doc.id,
						data: doc.data(),
					}))
				);
			});

		return unsubscribe;
	}, []);

	useLayoutEffect(() => {
		navigation.setOptions({
			title: "Chatsapp",
			headerStyle: { backgroundColor: "#fff" },
			headerTitleStyle: { color: "black" },
			headerTintColor: "black",
			headerLeft: () => (
				<View style={{ marginLeft: 20 }}>
					<TouchableOpacity activeOpacity={0.5} onPress={handleSignOutUser}>
						<Avatar
							rounded
							source={{
								uri: "https://cdn.iconscout.com/icon/free/png-256/logout-2477642-2061904.png",
							}}
						/>
					</TouchableOpacity>
				</View>
			),
			headerRight: () => (
				<View style={styles.headerRight}>
					<TouchableOpacity activeOpacity={0.5}>
						<AntDesign name="" size={24} color="black" />
					</TouchableOpacity>
					<TouchableOpacity
						activeOpacity={0.5}
						onPress={() => navigation.navigate("AddChat")}
					>
						<SimpleLineIcons name="plus" size={24} color="black" />
					</TouchableOpacity>
				</View>
			),
		});
	}, [navigation]);

	const enterChat = (id, chatName) => {
		navigation.navigate("Chat", {
			id,
			chatName,
		});
	};

	return (
		<SafeAreaView style={styles.container}>
			<StatusBar style="dark" />
			<Button onPress={() => setIsModalVisible(true)} title="Join Chat" />
			<ScrollView>
				{chats.map(({ id, data: { chatName } }) => (
					<CustomListItem
						key={id}
						id={id}
						chatName={chatName}
						enterChat={enterChat}
					/>
				))}
			</ScrollView>
			<Modal
				transparent={true}
				visible={isModalVisible}
				onDismiss={() => setIsModalVisible(false)}
			>
				<View
					style={{
						flex: 1,
						alignItems: "center",
						justifyContent: "center",
						backgroundColor: "rgba(0,0,0,0.7)",
					}}
				>
					<View
						style={{
							backgroundColor: "white",
							borderRadius: 8,
							width: "80%",
							height: 200,
							padding: 10,
							alignItems: "center",
							justifyContent: "center",
						}}
					>
						<Icon
							name="close"
							style={{ position: "absolute", top: 10, right: 10 }}
							size={20}
							onPress={() => {
								setChatRoomId("");
								setIsModalVisible(false);
							}}
						/>
						<Text
							style={{ marginBottom: 40, fontSize: 20, fontWeight: "bold" }}
						>
							Chat Room
						</Text>
						<Input
							placeholder="Enter Chat Room ID"
							value={chatRoomId}
							onChangeText={setChatRoomId}
							onSubmitEditing={handleJoinRoom}
							style={{ textAlign: "center" }}
						/>
						<Button
							disabled={!chatRoomId}
							onPress={handleJoinRoom}
							title="Join Chat Room"
							loading={isJoining}
						/>
					</View>
				</View>
			</Modal>
		</SafeAreaView>
	);
};

export default HomeScreen;

const styles = StyleSheet.create({
	container: {
		flex: 1,
		backgroundColor: "black",
	},
	headerRight: {
		flexDirection: "row",
		justifyContent: "space-between",
		width: 80,
		marginRight: 20,
	},

	container: {
		height: "100%",
	},
});
