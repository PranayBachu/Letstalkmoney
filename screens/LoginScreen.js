import React, { useState, useEffect } from "react";
import { StatusBar } from "expo-status-bar";
import { StyleSheet, View, ScrollView, Dimensions } from "react-native";
import { Button, Input, Image } from "react-native-elements";
import { auth } from "../firebase-services";
import {
	AdMobBanner,
	AdMobInterstitial,
	AdMobRewarded,
	setTestDeviceIDAsync,
} from "expo-ads-admob";

const logo = require("../assets/chatsapplogo.png");
const { height } = Dimensions.get("window");

const LoginScreen = ({ navigation }) => {
	const [email, setEmail] = useState("");
	const [password, setPassword] = useState("");
	const [isLoginLoading, setIsLoginLoading] = useState(false);
	const [isAnonymousLoginLoading, setIsAnonymousLoginLoading] = useState(false);

	const handleSignIn = async () => {
		try {
			setIsLoginLoading(true);
			await auth.signInWithEmailAndPassword(email, password);
		} catch (error) {
			alert(error);
			setIsLoginLoading(false);
		}
	};

	const handleAnonymousLogin = async () => {
		try {
			setIsAnonymousLoginLoading(true);
			await auth.signInAnonymously();
		} catch (error) {
			alert(error);
			setIsAnonymousLoginLoading(false);
		}
	};

	const setTestDeviceId = async () => {
		try {
			await setTestDeviceIDAsync("EMULATOR");
		} catch (error) {
			console.log(error);
		}
	};

	const interstatial = async () => {
		try {
			await AdMobInterstitial.setAdUnitID(
				"ca-app-pub-3940256099942544/1033173712"
			); // Test ID, Replace with your-admob-unit-id
			await AdMobInterstitial.requestAdAsync({ servePersonalizedAds: true });
			await AdMobInterstitial.showAdAsync();
		} catch (error) {
			console.log(error);
		}
	};
	const reward = async () => {
		try {
			await AdMobRewarded.setAdUnitID("ca-app-pub-3940256099942544/5224354917"); // Test ID, Replace with your-admob-unit-id
			await AdMobRewarded.requestAdAsync();
			await AdMobRewarded.showAdAsync();
		} catch (error) {
			console.log(error);
		}
	};

	useEffect(() => {
		setTestDeviceId();
		AdMobRewarded.addEventListener("rewardedVideoUserDidEarnReward", () => {
			alert("Ad Rewareded");
		});

		return () => AdMobRewarded.removeAllListeners();
	}, []);

	return (
		<ScrollView contentContainerStyle={styles.container}>
			<StatusBar style="light" />
			<Image source={logo} style={{ width: 300, height: 200 }} />
			<View style={styles.inputContainer}>
				<Input
					placeholder="Email"
					//autoFocus
					type="email"
					value={email}
					onChangeText={setEmail}
				/>
				<Input
					placeholder="Password"
					secureTextEntry
					type="password"
					value={password}
					onChangeText={setPassword}
					onSubmitEditing={handleSignIn}
				/>
			</View>
			<Button
				loading={isLoginLoading}
				containerStyle={styles.button}
				onPress={handleSignIn}
				title="Login"
			/>
			<Button
				backgroundColor="#ff0000"
				containerStyle={styles.button}
				loading={isAnonymousLoginLoading}
				onPress={handleAnonymousLogin}
				title="Anonymous Login"
				type="solid"
			/>
			<Button
				containerStyle={styles.button}
				onPress={() => navigation.navigate("Register")}
				title="Sign Up"
				type="outline"
			/>
			<View style={{ justifyContent: "flex-end", marginTop: 20 }}>
				<AdMobBanner
					bannerSize="banner"
					adUnitID="ca-app-pub-3940256099942544/6300978111" // Test ID, Replace with your-admob-unit-id
					servePersonalizedAds // true or false
					//onDidFailToReceiveAdWithError={this.bannerError}
				/>
				<Button
					buttonStyle={{ marginTop: 10 }}
					title="Interstatial Ad"
					onPress={interstatial}
				/>
				<Button
					buttonStyle={{ marginTop: 10 }}
					title="Rewarded Ad"
					onPress={reward}
				/>
			</View>
		</ScrollView>
	);
};

export default LoginScreen;

const styles = StyleSheet.create({
	container: {
		//flex: 1,
		alignItems: "center",
		justifyContent: "center",
		padding: 10,
		backgroundColor: "white",
		height,
	},
	inputContainer: {
		width: 300,
	},
	button: {
		marginTop: 10,
		width: 200,
	},
});
