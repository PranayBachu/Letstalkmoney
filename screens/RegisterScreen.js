import React, { useState } from "react";
import { StatusBar } from "expo-status-bar";
import { StyleSheet, View, ScrollView, Dimensions } from "react-native";
import { Button, Input, Text, Image } from "react-native-elements";
import { auth } from "../firebase-services";

const logo = require("../assets/chatsapplogo.png");
const { height } = Dimensions.get("window");

const RegisterScreen = () => {
	const [name, setName] = useState("");
	const [email, setEmail] = useState("");
	const [password, setPassword] = useState("");
	const [imageUrl, setImageUrl] = useState("");
	const [isRegistering, setIsRegistering] = useState(false);

	const handleRegister = async () => {
		try {
			setIsRegistering(true);
			const { user } = await auth.createUserWithEmailAndPassword(
				email,
				password
			);
			await user.updateProfile({
				displayName: name,
				photoURL:
					imageUrl ||
					"https://cencup.com/wp-content/uploads/2019/07/avatar-placeholder.png",
			});
		} catch (error) {
			alert(error.message);
			setIsRegistering(false);
		}
	};

	return (
		<ScrollView contentContainerStyle={styles.container}>
			<StatusBar style="light" />
			<Image source={logo} style={{ width: 200, height: 100 }} />
			<View style={styles.inputContainer}>
				<Input
					placeholder="Full Name"
					autoFocus
					type="text"
					value={name}
					onChangeText={(text) => setName(text)}
				/>
				<Input
					placeholder="Email"
					type="email"
					value={email}
					onChangeText={(text) => setEmail(text)}
				/>
				<Input
					placeholder="Password"
					type="password"
					secureTextEntry
					value={password}
					onChangeText={(text) => setPassword(text)}
				/>
				{/* <Input
					placeholder="Profile Picture URL (optional)"
					type="text"
					value={imageUrl}
					onChangeText={(text) => setImageUrl(text)}
					onSubmitEditing={handleRegister}
				/> */}
			</View>
			<Button
				raised
				title="Sign Up"
				containerStyle={styles.button}
				onPress={handleRegister}
				loading={isRegistering}
			/>
			<View style={{ height: 100 }} />
		</ScrollView>
	);
};

export default RegisterScreen;

const styles = StyleSheet.create({
	container: {
		flex: 1,
		alignItems: "center",
		justifyContent: "center",
		padding: 10,
		backgroundColor: "white",
		height,
	},
	button: {
		width: 200,
		marginTop: 10,
	},
	inputContainer: {
		width: 300,
		marginTop: 30,
	},
});
