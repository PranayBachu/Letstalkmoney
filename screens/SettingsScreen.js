import React, { useState } from "react";
import { StatusBar } from "expo-status-bar";
import { StyleSheet, View, ScrollView } from "react-native";
import { Button, Input, Text } from "react-native-elements";
import { auth, firebase } from "../firebase-services";

const SettingsScreen = ({ navigation }) => {
	const [name, setName] = useState(auth.currentUser.displayName);
	const [email, setEmail] = useState(auth.currentUser.email);
	const [password, setPassword] = useState("");
	const [isUpdating, setIsUpdating] = useState(false);

	const updateUserProfile = async () => {
		try {
			setIsUpdating(true);
			await auth.currentUser.updateProfile({
				displayName: name,
			});
			setIsUpdating(false);
			navigation.goBack();
		} catch (error) {
			alert(error);
			setIsUpdating(false);
		}
	};

	const upgradeFromAnonymous = async () => {
		try {
			setIsUpdating(true);
			const cred = firebase.auth.EmailAuthProvider.credential(email, password);
			const { user } = await auth.currentUser.linkWithCredential(cred);
			await user.updateProfile({
				displayName: name,
			});
			setIsUpdating(false);
			navigation.goBack();
		} catch (error) {
			alert(error);
			setIsUpdating(false);
		}
	};

	return (
		<ScrollView behavior="padding" contentContainerStyle={styles.container}>
			<StatusBar style="light" />
			<Text h1 style={{ marginBottom: 50 }}>
				Update Your Profile
			</Text>
			<View style={styles.inputContainer}>
				<Input
					placeholder="Full Name"
					autoFocus
					type="text"
					value={name}
					onChangeText={(text) => setName(text)}
				/>
				{auth.currentUser.isAnonymous && (
					<>
						<Input
							placeholder="Email"
							type="email"
							value={email}
							onChangeText={(text) => setEmail(text)}
						/>
						<Input
							placeholder="Password"
							type="password"
							secureTextEntry
							value={password}
							onChangeText={(text) => setPassword(text)}
						/>
					</>
				)}
				{/* <Input
          placeholder="Profile Picture URL (optional)"
          type="text"
          value={imageUrl}
          onChangeText={(text) => setImageUrl(text)}
          onSubmitEditing={updateUserProfile}
        /> */}
			</View>
			<Button
				//raised
				title="Update"
				containerStyle={styles.button}
				onPress={
					auth.currentUser.isAnonymous
						? upgradeFromAnonymous
						: updateUserProfile
				}
				loading={isUpdating}
			/>
			<View style={{ height: 100 }} />
		</ScrollView>
	);
};

export default SettingsScreen;

const styles = StyleSheet.create({
	container: {
		flex: 1,
		alignItems: "center",
		justifyContent: "center",
		padding: 10,
		backgroundColor: "white",
	},
	button: {
		width: 200,
		marginTop: 10,
	},
	inputContainer: {
		width: 300,
	},
});
