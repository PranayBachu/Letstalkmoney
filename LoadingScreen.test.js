import React from "react";
import renderer from "react-test-renderer";

import { LoadingScreen } from "./screens/LoadingScreen";

describe("<LoadingScreen />", () => {
  it("has 2 child", () => {
    const tree = renderer.create(<LoadingScreen />).toJSON();
    expect(tree.children.length).toBe(2);
    // console.log("the length is: ", tree.children.length);
  });
  it("renders correctly", () => {
    const tree = renderer.create(<LoadingScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
