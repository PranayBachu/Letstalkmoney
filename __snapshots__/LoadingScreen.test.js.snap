// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingScreen /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 200,
        "width": 200,
      }
    }
  />
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="small"
    style={
      Object {
        "marginTop": 10,
      }
    }
  />
</View>
`;
